### PR TITLE
Fix post-idle session UNKNOWN race condition, default old users to NTP, and bucket durations

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/1/137249556945/task/1214578665494291
Tech Design URL: N/A
CC: N/A

### Description

- Fix race condition causing ~70% of post-idle session wide events to complete as UNKNOWN: `sessionStarted()` now synchronously cleans orphaned flows before creating a new one, and `completionDecision(for:)` returns `.keepPending` so the async resume path cannot race with new flow creation
- Default old iPhone users (with no stored preference) to New Tab Page after idle, matching new user behavior. Users who explicitly chose "Last Used Tab" in Settings keep that preference. iPad continues to default to Last Used Tab
- Bucket session duration and time-to-first-interaction to match Android's schema (privacy requirement): values are now bucketed strings (`0`, `1000`, `5000`, `10000`, `30000`, `60000`, `300000`, `600000` ms) instead of raw integers

### Testing Steps

**Setup (run once after installing the app):**

```bash
# Find your app container (use the correct bundle ID for your build)
# Alpha: com.duckduckgo.mobile.ios.alpha
# Debug: com.duckduckgo.mobile.ios
BUNDLE_ID="com.duckduckgo.mobile.ios.alpha"
CONTAINER=$(xcrun simctl get_app_container booted $BUNDLE_ID data)
PLIST="$CONTAINER/Library/Application Support/AppKeyValueStore.plist"

# Set idle threshold to 5 seconds for quick testing
xcrun simctl spawn booted defaults write $BUNDLE_ID idle-threshold-override -float 5
```

**Scenario 1: Old user with no stored preference (should see NTP)**

```bash
xcrun simctl terminate booted $BUNDLE_ID
plutil -remove idle-return-after-inactivity-option "$PLIST" 2>/dev/null
plutil -replace idle-return-new-user -bool false "$PLIST"
xcrun simctl launch booted $BUNDLE_ID
```

1. Background the app for 5+ seconds, then reopen
2. Verify NTP is shown after idle
3. Go to Settings > General > Show on App Launch — verify "New Tab Page" is selected

**Scenario 2: User who explicitly chose Last Used Tab (should see LUT)**

```bash
xcrun simctl terminate booted $BUNDLE_ID
plutil -replace idle-return-after-inactivity-option -string "lastUsedTab" "$PLIST"
xcrun simctl launch booted $BUNDLE_ID
```

1. Background the app for 5+ seconds, then reopen
2. Verify Last Used Tab is shown after idle
3. Go to Settings > General > Show on App Launch — verify "Last Used Tab" is selected

**Scenario 3: New user (should see NTP)**

```bash
xcrun simctl terminate booted $BUNDLE_ID
plutil -remove idle-return-after-inactivity-option "$PLIST" 2>/dev/null
plutil -replace idle-return-new-user -bool true "$PLIST"
xcrun simctl launch booted $BUNDLE_ID
```

1. Background the app for 5+ seconds, then reopen
2. Verify NTP is shown after idle

Scenario 4: Race condition fix (no UNKNOWN accumulation)

Kill the app mid-session (force quit while on NTP/LUT after idle)
Relaunch the app
Trigger another idle session
Verify the orphaned session completes as UNKNOWN with app_terminated reason and the new session starts cleanly (no double-completion)
Smoke test we get the wide pixels as expected

### Impact and Risks
**Impact Level: Medium**

#### What could go wrong?
- Old users who were accustomed to seeing their last tab after idle will now see NTP (intentional behavior change, users can switch back in Settings)
- Duration values in existing dashboards will shift from exact ms to bucketed values (dashboard update planned separately to COALESCE both keys)

### Quality Considerations
- Race condition fix is defensive: orphan cleanup runs synchronously before new flow creation
- Bucketing thresholds match Android exactly for cross-platform consistency
- All boundary cases tested (0ms, 999ms, 1000ms, etc.)
- iPad behavior unchanged (always defaults to Last Used Tab)

---

> [!NOTE]
> **Medium Risk**
> Medium risk due to changes in post-idle wide-event completion timing (orphan cleanup and completion decisions) and a behavior change in default after-idle tab selection on iPhone; could affect analytics and user-facing defaults.
> 
> **Overview**
> **Post-idle session wide-event reliability:** `sessionStarted()` now synchronously completes any orphaned `PostIdleSessionWideEventData` flows as `.unknown(reason: "app_terminated")` *before* starting a new flow, and `completionDecision(for:)` no longer completes on `.appLaunch` (returns `.keepPending`) to avoid resume-time races that were producing UNKNOWN completions.
> 
> **After-inactivity default change:** When no preference is stored, iPhone now defaults to `.newTab` (and persists it) regardless of `idleReturnNewUser`, while iPad continues to default to `.lastUsedTab`; tests updated/expanded accordingly.
> 
> **Privacy/metrics schema update:** Post-idle session duration and time-to-first-interaction parameters switch from raw millisecond integers to **bucketed string** values (new `*_ms_bucketed` keys), with matching PixelDefinitions updates and new boundary-condition tests.
> 
> **Test utilities:** `WideEventMock` now tracks completed/discarded IDs so `getFlowData`/`getAllFlowData` exclude finished flows, supporting the new orphan-cleanup tests.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1ced6eba7b557635c5e9cb725f733a5dec597703. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: iOS/DuckDuckGo/AppLifecycle/PostIdleSessionInstrumentation.swift [hunk 1]
```diff
@@ -131,6 +137,15 @@ final class DefaultPostIdleSessionInstrumentation: PostIdleSessionInstrumentatio
 
     // MARK: - Helpers
 
+    private func completeOrphanedFlows() {
```
Comment (Bunn) [inline iOS/DuckDuckGo/AppLifecycle/PostIdleSessionInstrumentation.swift:140]: Would be good to add test cases covering this flow

File: iOS/DuckDuckGoTests/PostIdleSessionWideEventDataTests.swift [hunk 2]
```diff
@@ -157,29 +181,28 @@ struct PostIdleSessionWideEventDataTests {
     // MARK: - Completion decision
```
Comment (Bunn) [inline iOS/DuckDuckGoTests/PostIdleSessionWideEventDataTests.swift:182]:
```suggestion
 // MARK: - Orphan cleanup

    @available(iOS 16, *)
    @Test("sessionStarted completes orphaned flows as UNKNOWN with app_terminated", .timeLimit(.minutes(1)))
    func sessionStartedCompletesOrphansAsAppTerminated() {
        let (sut, wideEvent, _) = makeSUT()
        // Simulate an orphan left in storage by a previous app lifecycle.
        let orphan = PostIdleSessionWideEventData(surface: .ntp)
        wideEvent.startFlow(orphan)

        sut.sessionStarted(surface: .ntp)

        let orphanCompletion = wideEvent.completions.first {
            ($0.0 as? PostIdleSessionWideEventData)?.globalData.id == orphan.globalData.id
        }
        guard let orphanCompletion else {
            Issue.record("Expected the orphan to be completed")
            return
        }
        if case .unknown(let reason) = orphanCompletion.1 {
            #expect(reason == PostIdleSessionWideEventData.appTerminatedReason)
        } else {
            Issue.record("Expected .unknown status, got \(orphanCompletion.1)")
        }
    }

    @available(iOS 16, *)
    @Test("Orphan cleanup happens before the new flow is started", .timeLimit(.minutes(1)))
    func orphanCleanupHappensBeforeNewFlowStarts() {
        let (sut, wideEvent, _) = makeSUT()
        let orphan = PostIdleSessionWideEventData(surface: .ntp)
        wideEvent.startFlow(orphan)

        sut.sessionStarted(surface: .lut)

        // started[0] is the seeded orphan; started[1] must be the new LUT flow,
        // and the orphan completion must already be recorded by the time it lands.
        #expect(wideEvent.started.count == 2)
        #expect((wideEvent.started[1] as? PostIdleSessionWideEventData)?.surface == .lut)
        let orphanCompleted = wideEvent.completions.contains {
            ($0.0 as? PostIdleSessionWideEventData)?.globalData.id == orphan.globalData.id
        }
        #expect(orphanCompleted)
    }

    @available(iOS 16, *)
    @Test("Multiple orphans are all completed", .timeLimit(.minutes(1)))
    func multipleOrphansAllCompleted() {
        let (sut, wideEvent, _) = makeSUT()
        let firstOrphan = PostIdleSessionWideEventData(surface: .ntp)
        let secondOrphan = PostIdleSessionWideEventData(surface: .lut)
        wideEvent.startFlow(firstOrphan)
        wideEvent.startFlow(secondOrphan)

        sut.sessionStarted(surface: .ntp)

        let unknownCompletions = wideEvent.completions.filter {
            if case .unknown = $0.1 { return true } else { return false }
        }
        #expect(unknownCompletions.count == 2)
    }

    @available(iOS 16, *)
    @Test("sessionStarted with no orphans does not produce spurious completions", .timeLimit(.minutes(1)))
    func sessionStartedWithoutOrphansProducesNoCompletions() {
        let (sut, wideEvent, _) = makeSUT()
        sut.sessionStarted(surface: .ntp)
        #expect(wideEvent.completions.isEmpty)
    }
```

**Review by Bunn — APPROVED** [review]
LGTM, one suggestion
